Make package architecture

kelshmo opened the pull request:
This PR changes the dir structure to follow package convention (e.g. use of `inst/`) and adds `man/` documentation. Currently, `devtools::check()` passes without errors.

- One of the outputs of the workflow is a markdown which I have stored in the `inst/` directory. This might be replaced at a later date with `knitr` calls directly from the plan.
- `config.yml` is ignored by the R build. Taking the approach of the [dccvalidator](https://sage-bionetworks.github.io/dccvalidator/articles/customizing-dccvalidator.html) and communicating about the config via documentation. 
- The `initialize.R` file is called from the top-level directory which is non-standard. This is how the `drake` workflow repo is recommended to be structured. I need to consider how to modify this if I wish to wrap the entire `drake` workflow as a package.
- I was recommended by `devtools::check()` to `  Consider adding
    importFrom("utils", "type.convert")` - I need to research how to import a single function from a package and may want to apply this to other cases.
- Maybe need to re-prioritize `Imports` and `Depends` in DESCRIPTION. If I understand it correctly, Depends is of highest importance. For this package `drake` would fall under that level of importance.
- I'm returning many `Undocumented arguments in documentation object` warnings which suggests I may be calling `@inheritParams` incorrectly.

There are some warnings from `devtools::check()` that I'm not sure how to deal with:
- `  boxplot_vars: no visible global function definition for ‘%>%’` I'm not sure the syntax to call the magrittr
- `  boxplot_vars: no visible global function definition for ‘aes’` Is is redundant to fix this to `ggplot2::ggplot(ggplot2::aes(x = x_var, y = value))`?
- `  convert_geneids: no visible binding for global variable ‘ids’` for a tibble column name `convert_geneids <- function(count_df) {
    geneids <- tibble::tibble(ids = rownames(count_df))`

Overall, need to critically consider the approach of wrapping the entire drake workflow as a package. Doing so obscures the workflow steps from the end-user. It may make more sense to build a package for the underlying functions and separately call the workflow. 

All that said, the package installed succesfully!

I started to write "This package.. " in the DESCRIPTION then I remembered the unspoken **rules**.

karawoo commented on DESCRIPTION:
```diff
@@ -1,15 +1,41 @@
-License: What license it uses
+           email = "kelsey.montgomery@sagebase.org",
+           comment = c(ORCID = "0000-0002-0126-8944"))
+Description: This workflow normalizes counts, performs model selection from known 
+covariates using multivariate regression, automates outlier analysis using a 
+customizable threshold and performs differential expression analysis for repeated
+measures.
```
Lines 11-13 need to be indented 4 spaces

karawoo commented on DESCRIPTION:
```diff
@@ -1,15 +1,41 @@
 Suggests: 
     testthat (>= 2.1.0)
+RoxygenNote: 7.1.0
+Imports: 
```
```suggestion
biocViews:
Imports: 
```
trick to help R find dependencies from bioconductor when installing the package

karawoo commented on DESCRIPTION:
```diff
@@ -1,15 +1,41 @@
+RoxygenNote: 7.1.0
+Imports: 
+    config,
+    CovariateAnalysis,
```
Where is this package installed from? I can't find it

kelshmo replied:
I forgot it has to be specified as a remote if it is from Github.

karawoo commented on DESCRIPTION:
```diff
@@ -1,15 +1,41 @@
+    IRanges,
+    glue,
+    tidyr,
+    ggplot2
```
```suggestion
    ggplot2,
    utils
```

You are using `utils::type.convert()`, so utils needs to be listed as a dependency

karawoo commented on DESCRIPTION:
```diff
@@ -1,15 +1,44 @@
     testthat (>= 2.1.0)
+RoxygenNote: 7.1.0
+biocViews:
+    limma,
```
The bioconductor packages don't go under the biocViews line, it's just there to tell R that it should look at both CRAN and bioconductor when installing the packages that are in Depends/Imports/etc.

kelshmo commented:
@karawoo Thanks for your quick response! 

- Fixed DESCRIPTION. 
- Fixed @inheritParams.
- `rlang .data` fixed all cases except for 3 that I will address in #42.

The `R/packages.R` is an artifact of following the `drake` workflow set-up as recommended. I think it makes sense to remove it as the package itself will take care of loading package dependencies. The recommended structure for drake is `R/functions.R`, `R/packages.R` and `R/plan.R`. 

This gets at whether I should wrap the workflow as a package. Right now the workflow is called in  `drake_plan` in the `plan.R` file. That is not getting surfaced in the `sageseqr` package as far as I can tell.

karawoo commented on DESCRIPTION:
```diff
@@ -1,15 +1,46 @@
 Package: sageseqr
-Title: What the Package Does (One Line, Title Case)
+Title: Identify differentially expressed genes counts data.
```
```suggestion
Title: Identify differentially expressed genes counts data
```
R CMD check doesn't like periods at the end of the package title 🙄

karawoo commented on DESCRIPTION:
```diff
@@ -1,15 +1,46 @@
+biocViews:
+Imports:
+    biomaRt,
+    config,
```
```suggestion
    config,
    CovariateAnalysis,
```

The DESCRIPTION still needs to have CovariateAnalysis under Imports in addition to having 

```
Remotes:
    th1vairam/CovariateAnalysis@dev
```

Imports tells R which packages are required, and Remotes tells it where to find them if not CRAN.

karawoo commented on R/functions.R:
```diff
@@ -73,32 +79,33 @@ clean_covariates <- function(md, factors, continuous) {
+#'@importFrom rlang .data
 #'
 #'@export
 #'@return
```
This line is causing a NOTE. `@return` is for documenting what the function returns, so to prevent the NOTE you could either remove this line or add a description of what is returned by the function.

kelshmo commented:
I updated the code based on your suggestions and added `initialize.R` to the build ignore file. I do think I will wrap the plan as a function and also expose it through documentation (via `pkgdown` maybe) in case a user wants to manipulate the plan and add steps. There are so many subtle nuances to making a package - thank you for transferring your knowledge!!!
